lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -706,69 +700,21 @@ impl SegmentEntry for ProxySegment {
             "Taking a snapshot of a proxy segment into {:?}",
             snapshot_dir_path
         );
-        // extra care is needed to capture outstanding deleted points
-        let deleted_points_guard = self.deleted_points.read();
-        let wrapped_segment_arc = self.wrapped_segment.get();
-        let wrapped_segment_guard = wrapped_segment_arc.read();
-
-        // stable copy of the deleted points at the time of the snapshot
-        let deleted_points_copy = deleted_points_guard.clone();
-
-        // parse wrapped segment id from it's data path
-        let wrapped_segment_id = wrapped_segment_guard
-            .data_path()
-            .file_stem()
-            .unwrap()
-            .to_str()
-            .unwrap()
-            .to_string();
-
-        // create temporary dir for operations on wrapped segment data
-        let tmp_dir_path = snapshot_dir_path.join(format!("segment_copy_{}", Uuid::new_v4()));
-        create_dir_all(&tmp_dir_path)?;
-
-        // snapshot wrapped segment data into the temporary dir
-        let archive_path = wrapped_segment_guard.take_snapshot(&tmp_dir_path)?;
 
-        // restore a copy of the wrapped segment data into the temporary dir
-        Segment::restore_snapshot(&archive_path, &wrapped_segment_id)?;
+        let archive_path = {
+            let wrapped_segment_arc = self.wrapped_segment.get();
+            let wrapped_segment_guard = wrapped_segment_arc.read();
 
-        // build a path to a copy of the wrapped segment data
-        let full_copy_path = tmp_dir_path.join(&wrapped_segment_id);
+            // snapshot wrapped segment data into the temporary dir
+            wrapped_segment_guard.take_snapshot(snapshot_dir_path)?
+        };
 
         // snapshot write_segment
         let write_segment_rw = self.write_segment.get();
         let write_segment_guard = write_segment_rw.read();
 
         // Write segment is not unique to the proxy segment, therefore it might overwrite an existing snapshot.
         write_segment_guard.take_snapshot(snapshot_dir_path)?;
-        // guaranteed to be higher than anything in wrapped segment and does not exceed WAL at the same time
-        let write_segment_version = write_segment_guard.version();
-
-        // unlock deleted_points as we have a stable copy
-        drop(wrapped_segment_guard);
-        drop(deleted_points_guard);
-
-        // load copy of wrapped segment in memory
-        let mut in_memory_wrapped_segment = load_segment(&full_copy_path)?.ok_or_else(|| {
-            OperationError::service_error(format!(
-                "Failed to load segment from {:?}",
-                full_copy_path
-            ))
-        })?;
-
-        // remove potentially deleted points from wrapped_segment
-        for deleted_point in deleted_points_copy {
-            in_memory_wrapped_segment.delete_point(write_segment_version, deleted_point)?;
```

Comment:
We were previously deleting explicitly the deleted points in the wrapped segment based on the proxy information.
How is the the deduplication strategy handling those deleted points in the proxy?

Reply:
deleted points are handled by records in WAL. The operations in WAL will be re-applied to all segments and if we removed something - it will also be removed on recovery.